src/solvers/refinement/string_refinement.cpp

```diff
@@ -1499,24 +1787,28 @@ static std::vector<exprt> sub_arrays(const exprt &array_expr)
 /// add to the index set all the indices that appear in the formula and the
 /// upper bound minus one
 /// \par parameters: a string constraint
-void string_refinementt::add_to_index_set(const exprt &s, exprt i)
+static void add_to_index_set(
+  std::map<exprt, std::set<exprt>> &index_set,
+  std::map<exprt, std::set<exprt>> &current_index_set,
+  const namespacet &ns,
+  const exprt &s,
+  exprt i)
```

Comment:
always make a copy?

Reply:
Yes copying is necessary, as `simplify(i, ns)` requires non-const `exprt`.